RpcServer/RpcServer/RpcServer.cs

```diff
@@ -0,0 +1,744 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Net.Security;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.ResponseCompression;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.Extensions.DependencyInjection;
+using Neo.IO.Json;
+
+namespace Neo.Plugins.RpcServer
+{
+    public class RpcServer : Plugin, IRpcServer
+    {
+        #region Variables
+
+        private const int MaxPostValue = 1024 * 1024 * 2;
+
+        private IWebHost _host;
+        private RpcServerSettings _settings;
+        private IDictionary<string, IDictionary<string, RcpTargetAndMethod>> _operations = new Dictionary<string, IDictionary<string, RcpTargetAndMethod>>();
+        private IDictionary<Type, Func<HttpContext, object>> _specialParameterInjectors = new Dictionary<Type, Func<HttpContext, object>>();
+        private List<Func<IRpcOperationPayload, IRpcOperationPayload>> _requestInterceptors = new List<Func<IRpcOperationPayload, IRpcOperationPayload>>();
+        private List<Func<JObject, JObject>> _responseInterceptors = new List<Func<JObject, JObject>>();
+
+        #endregion
+
+        public override string Name => "RpcServer";
+
+        public override void Configure()
+        {
+            _settings = new RpcServerSettings(GetConfiguration());
+            InjectSpecialParameter(ctx => ctx);
+            InjectSpecialParameter(ctx => _settings);
+
+            foreach (IRpcPlugin plugin in Plugin.RpcPlugins.ToList())
+            {
+                plugin.BeforeStartServer(this);
+            }
+
+            Start();
+        }
+
+        private JObject CreateErrorResponse(bool isNewVersion, string id, int code, string message, string stacktrace = null)
+        {
+            var response = CreateResponse(isNewVersion, id);
+            response["error"] = new JObject
+            {
+                ["code"] = code,
+                ["message"] = message
+            };
+
+#if DEBUG
+            if (stacktrace != null)
+            {
+                response["error"]["data"] = stacktrace;
+            }
+#endif
+
+            return CallResponseInterceptors(response);
+        }
+
+        private JObject CreateResponse(bool isNewVersion, string id)
+        {
+            var response = new JObject();
+
+            if (isNewVersion)
+            {
+                response["restResponse"] = "1.0";
+            }
```

Comment:
What's this?

Reply:
if the request comes with the REST format we can't answer with a jsonRpc version, so it is setting a "restResponse" version

Reply:
I think it's not a good idea to mix JSON-RPC with REST.

Reply:
Maybe we need a plugin for Http and then a plugin for RPC and another for Rest.

Reply:
I can easily separate http, json-rpc and rest, this is the best solution?
But without json-rpc or rest, there will be no neo methods implemented.

Reply:
Ok, it was not easy, but now we have an HttpServer plugin, a RpcInterceptor and RestInterceptor.
They are all in the same package for now but I will separate them.